### DISCUSSION TITLE
Setting timeout and backoff/retry for Sentry SDK

### DISCUSSION THREAD

**Discussion by jjblumenfeld** [getsentry/sentry-python#1945] [Q&A] [answered]
Hi,
Is it possible to set timeout and retry behavior for the Sentry client when initializing it?
Background is that we are using Sentry self-hosted and we have had issues where it will be unavailable for periods of time. During that time our Django (gunicorn)web-workers may get stuck trying to reach Sentry and we'd rather just have those requests fail rather than timeout after the default timeout and then retry with the default backoff.

Is there an easy way to set this when initializing the client?

I have found this stackoverflow discussion, but I'm not sure if there is a better way.
https://stackoverflow.com/questions/54592803/sentry-sdk-timeout-setting

**Comment by sl0thentr0py** [comment, marked as answer]
HI @jjblumenfeld, we don't support this out of the box but you can [write your own `Transport`](https://github.com/getsentry/sentry-python/discussions/1484) which does this. You can derive from [`HttpTransport`](https://github.com/getsentry/sentry-python/blob/2c8d27788c7e78a2e24e264d0e2d2f221e157658/sentry_sdk/transport.py#L133) and just patch `_send_request` to add some backoff/retry logic.
